### PR TITLE
handle kserve in global model serving page

### PR DESCRIPTION
**Pull request by christianvogt**
Addresses most of #1948 but not everything since the edit modal for kserve isn't yet available. For now the edit action for a kserve inference service is disabled.
Also does not expose GRPC url in table (out of scope).
Does not address deletion of `ServingRuntime` as this will be addressed by `ownerReferences` when we create the kserve `ServingRuntime` and `InferenceService` - related to https://github.com/opendatahub-io/odh-dashboard/issues/1998

## Description

Updates the model serving table to display column for `Serving runtime` and removes the `Model server` column.
Changed title of the global model serving page to `Deployed models`

image

fyi @vconzola 

## How Has This Been Tested?

- Create a model server: `ServingRuntime`
- Deploy a model: `InferenceService`
- Edit the `InferenceService` and change annotation `serving.kserve.io/deploymentMode` to a value other than `ModelMesh`
- Go to the `Model Serving` page using the left navigation
- Observe the changes to the table
- Click the kebab menu for your `InferenceService` and observe that `Edit` is disabled
- Use the `Delete` action
- Observe the `InferenceService` is deleted (Note that until the kserve creation flows are complete, the `ServingRuntime` will not be deleted as we will rely on `ownerReferences`)

## Test Impact

Added some unit tests.

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [x] Included any necessary screenshots or gifs if it was a UI change.
- [x] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: frontend/src/pages/modelServing/screens/global/DeleteInferenceServiceModal.tsx [hunk 1]
```diff
@@ -29,16 +33,24 @@ const DeleteInferenceServiceModal: React.FC<DeleteInferenceServiceModalProps> =
...
-            inferenceService.metadata.name,
-            inferenceService.metadata.namespace,
-          )
+          Promise.all([
+            deleteInferenceService(
+              inferenceService.metadata.name,
+              inferenceService.metadata.namespace,
+            ),
+            servingRuntime
+              ? deleteServingRuntime(
```
Comment (lucferbux) [inline frontend/src/pages/modelServing/screens/global/DeleteInferenceServiceModal.tsx:48]: Be aware that there are several resources to delete with the serving runtime too, such as the rolebinding, secrets and more, I think we could cover this in https://github.com/opendatahub-io/odh-dashboard/issues/1998 but this logic should be similar to https://github.com/opendatahub-io/odh-dashboard/blob/main/frontend/src/pages/modelServing/screens/projects/DeleteServingRuntimeModal.tsx#L50

**Review by lucferbux — CHANGES_REQUESTED** [review]
Ok, so far the changes look good, we need to review the logic to delete the `ServingRuntime` as we'll need to delete other resources too and there's a new version of the mockups, if you can take a look would be awesome!

File: frontend/src/pages/modelServing/screens/global/DeleteInferenceServiceModal.tsx (hunk 1 above)
Comment (christianvogt) [reply frontend/src/pages/modelServing/screens/global/DeleteInferenceServiceModal.tsx:48]: The complex deletion of the ServingRuntime should be handled in #1998.

That being said, I am deleting two related resources here and I wonder if we should rely on creating an ownerReference between InferenceService and ServingRuntime for kserve. If so then I don't need to delete both resources here.

**Comment by christianvogt** [comment]
@lucferbux I update the PR to include the latest UX change for the project label.

I've also removed the deletion change I initially made in lieu of https://github.com/opendatahub-io/odh-dashboard/issues/1998. As per our slack conversation, we will utilize `ownerReferences` to perform the clean up of related resources.

File: frontend/src/pages/modelServing/screens/global/InferenceServiceProject.tsx [hunk 2]
```diff
@@ -27,7 +27,22 @@ const InferenceServiceProject: React.FC<InferenceServiceProjectProps> = ({ infer
 
   const project = modelServingProjects.find(byName(inferenceService.metadata.namespace));
 
-  return <>{project ? getProjectDisplayName(project) : 'Unknown'}</>;
+  return (
+    <>
+      {project ? (
+        <>
+          {getProjectDisplayName(project)}{' '}
```
Comment (lucferbux) [inline frontend/src/pages/modelServing/screens/global/InferenceServiceProject.tsx:34]: We could give a little bit of padding, but that can be a follow up enhancement and we can discuss it later with UX

**Review by lucferbux — APPROVED** [review]
/lgtm

I've left a little nit that we can change later on, but we can unblock this

**Comment by vconzola** [comment]
LGTM

**Pull request merged**